Create a character creator with choses for race and class

Rafaelorr opened the pull request.

Rafaelorr committed:
remove the Dryad race

Rafaelorr committed:
Add types for class and race part of the code

Rafaelorr committed:
add another file for the random character genartor

Rafaelorr committed:
add bookmarks extension bookmarks for all races

Rafaelorr committed:
removed succubus in the generator

Rafaelorr committed:
removed Mousefolk in the generator

Rafaelorr committed:
deleted homebrew races from the random selection

Rafaelorr committed:
remove homebrew races from creation.py

Rafaelorr committed:
Let the user chose their (sub)race in Creation.py

Rafaelorr committed:
Let the user choose their subclass in creation.py

Rafaelorr committed:
Add bookmarks for all the races in creation.py

Rafaelorr commented:
@maxbotsis Please merge if you have time

maxbotsis commented on Dungeons_and_Dragons_Character_Creation.py:
```diff
@@ -28,153 +28,139 @@ def Normal(Min, Max): # Not exactly sure how this one is working, but it gives a
+                  Aasimar, Bugbear, Dragonborn, Dwarf, Elf, Firbolg, Genasi, Bith, Gnome,
+                  Goblin, Goliath, Hobgoblin, Half-Elf, Halfing, Half-Orc, Human, Juiblexian,
+                  Kender, Kenku, Kobold, Lizardfolk, Orc, Tabaxi, Tiefling,
+                  Tortle, Triton, Yuan-Ti, Pureblood
```
"Yuan-Ti Pureblood" is one race. Comma is in error here.

maxbotsis replied:
I'm actually going to pull this to fix it along with some other minor typos. I don't know how to use GitHub properly, so ignore this comment, see above points.

maxbotsis commented:
On second pass, I decided to take your changes and then fix it up with some improved code with a friend. I'll name you in the notes once it's all wrapped up.

Rafaelorr commented:
For clarification:
- bookmarks.json is used for a bookmarking extension use in vscode
- I deleted homebrew races because I couldn't find the sources or the links were dead
- You can only pick your (sub)class and race in generator.py
-  I added static typing to make the code more readable

maxbotsis commented:
I'm doing a complete refactor of this to make it follow better coding practices (a lot of my old code is very poor by PEP8 standards), so I'm going to close this PR for now. I should be done in about a week or two and I'll try to add in the functionality of choosing (sub)classes and race similar to how you did it.

The changes on my end are going to be quite major, so I can't merge your changes in without breaking things. But thank you for inspiring me to revisit this script.

Rafaelorr commented:
No problem
